perf/runner/src/index.ts

```diff
@@ -0,0 +1,150 @@
+                    clientPublicIP,
+                    serverPublicIP,
+                    uploadBytes: 0,
+                    downloadBytes: 10 * 1024 * 1024,
+                    nTimes: 1,
+                }),
+                comparisons: [],
+            },
+            {
+                name: "Single Connection 1 byte round trip latency",
+                unit: "s",
+                results: runBenchmarkAcrossVersions({
+                    clientPublicIP,
+                    serverPublicIP,
+                    uploadBytes: 1,
+                    downloadBytes: 1,
+                    nTimes: 10,
+                }),
+                comparisons: [],
+            }
+        ],
+    };
+
+    console.log(JSON.stringify(benchmarkResults, null, 2));
+}
+
+interface ArgsRunBenchmarkAcrossVersions {
+    clientPublicIP: string;
+    serverPublicIP: string;
+    uploadBytes: number,
+    downloadBytes: number,
+    nTimes: number,
+}
+
+function runBenchmarkAcrossVersions(args: ArgsRunBenchmarkAcrossVersions): Result[] {
+    const results: Result[] = [];
+    for (const version of versions) {
+        for (const transportStack of version.transportStacks) {
+            const latencies = runBenchmark({
+                clientPublicIP: args.clientPublicIP,
+                serverPublicIP: args.serverPublicIP,
+                dockerImageId: version.containerImageID,
+                transportStack: transportStack,
+                uploadBytes: args.uploadBytes,
+                downloadBytes: args.downloadBytes,
+                nTimes: args.nTimes,
+            });
+
+            results.push({
+                result: latencies.latencies,
+                implementation: "",
+                version: version.id,
+                transportStack: transportStack,
+            });
+        }
+    };
+
+    return results;
+}
+
+interface ArgsRunBenchmark {
+    clientPublicIP: string;
+    serverPublicIP: string;
+    dockerImageId: string;
+    transportStack: string,
+    uploadBytes: number,
+    downloadBytes: number,
+    nTimes: number,
+}
+
+interface Latencies {
+    latencies: number[];
+}
+
+
+function runBenchmark(args: ArgsRunBenchmark): Latencies {
+    let serverAddress: string;
+
+    switch (args.transportStack) {
+        case 'tcp':
+            serverAddress = `/ip4/${args.serverPublicIP}/tcp/4001`;
+            break;
+        case 'quic-v1':
+            serverAddress = `/ip4/${args.serverPublicIP}/udp/4001/quic-v1`;
+            break;
+        default:
+            console.error("Unsupported transport stack ${args.transportStack}");
+            process.exit(1);
+
+    }
+
+    const binFlags = `--server-address ${serverAddress} --upload-bytes ${args.uploadBytes} --download-bytes ${args.downloadBytes} --n-times ${args.nTimes}`
+    // TODO Take docker hub repository from version.ts
+    const dockerCMD = `docker run --rm --entrypoint perf-client mxinden/libp2p-perf@sha256:${args.dockerImageId} ${binFlags}`
```

Comment:
I assume we don't want the entrypoint or the `mxinden/libp2p-perf` here, right?

Reply:
Nit pick, the entrypoint is `perf-client`. Either way, yes, I will change the Rust image to use `perf-client` by default. In other words, for the JS and Go image, no `--entrypoint` should be needed.